test/object-store/sync/flx_sync.cpp

```diff
@@ -4688,6 +4688,33 @@ TEST_CASE("flx: fatal errors and session becoming inactive cancel pending waits"
     check_status(state);
 }
 
+TEST_CASE("flx: pause and resume bootstrapping at query version 0", "[sync][flx][baas]") {
+    FLXSyncTestHarness harness("flx_pause_resume_bootstrap");
+    SyncTestFile triggered_config(harness.app()->current_user(), harness.schema(), SyncConfig::FLXSyncEnabled{});
+    auto [interrupted_promise, interrupted] = util::make_promise_future<void>();
+    int download_message_received_count = 0;
+    triggered_config.sync_config->on_sync_client_event_hook =
+        [promise = util::CopyablePromiseHolder(std::move(interrupted_promise)), &download_message_received_count](
+            std::weak_ptr<SyncSession> weak_sess, const SyncClientHookData& data) mutable {
+            auto sess = weak_sess.lock();
+            if (!sess || data.event != SyncClientHookEvent::DownloadMessageReceived) {
+                return SyncClientHookAction::NoAction;
+            }
+
+            if (download_message_received_count == 0) {
+                sess->pause();
+                sess->resume();
+            }
+            else {
+                promise.get_promise().emplace_value();
+            }
+            ++download_message_received_count;
+            return SyncClientHookAction::NoAction;
+        };
+    auto realm = Realm::get_shared_realm(triggered_config);
+    interrupted.get();
```

Comment:
Don't we need to wait for the bootstrap to finish here for the test to really validate whether this fix has worked?

Reply:
As a matter of fact the test does finish, but it's not immediately obvious why. I think it helps the readability if we fulfil the promise after the message is integrated (I'll add some comments too).

Reply:
Should we check here that the subscription set version is zero and the subscription set state is Complete?

Reply:
will do

Reply:
thinking about it more, do we even need the promise in that case. could the test outside the event hook just wait for the subscription set to be complete?

Reply:
no. the subscription set is complete as part of the first session, but the second session and its session wrapper used to read the state of the subscription store at the wrong time.